### ISSUE TITLE
[HELP]How to record truely  in .wav format audio

### ISSUE THREAD

**Issue by islom9797** [Canardoux/flutter_sound#973]
How to record in .wav format audio
## Here is my question :

    Directory tempDir = await getApplicationDocumentsDirectory();
   String path = tempDir.path + '/audio.wav';
   mRecorder?.startRecorder(toFile: path, codec: Codec.pcm16WAV)
 this source dont give any error but when I stop record value(path of recorded audio) is empty

  await mRecorder!.stopRecorder().then((value) async {
      print('value on stopRecord $value');

**Comment by sofakeer** [comment]
it work

**Comment by RitchieP** [comment]
@islom9797 Hi, have you happen to make this work? I'm also facing the same issue

**Comment by muneebahmad0600** [comment]
Having same issue.
`  void startRecorder() async {
    if (_mRecorder!.recorderState == RecorderState.isPaused) {
      resumeRecorder();
      return;
    }
    Directory tempDir = await getApplicationDocumentsDirectory();
    _mRecorder!
        .startRecorder(
      toFile: '${tempDir.path}/audio.wav',
      codec: Codec.pcm16WAV,
      audioSource: theSource,
    )
        .then((value) {
      setState(() {
        _isRecording = true;
        context.read<RecordingProvider>().toggleTimer();
      });
    });
  }


  void stopRecorder() async {
    await _mRecorder!.stopRecorder().then((value) {
      context.read<RecordingProvider>().pauseTimer();
      setState(() {
        _recordingPath = value.toString();
        _isRecording = false;
        Timer(const Duration(milliseconds: 500), () {
          NavigationService.pushReplacement(
            UploadRecordingScreen(recordingPath: _recordingPath),
          );
        });
      });
    });
  }`
  
  value is empty string when recording is stopped.

**Comment by Larpoux** [comment]
Actually, I know that there is a bug in Flutter Sound when the App tries to use the File Path returned by stopRecorder().

This is perhaps the reason for your issue.
I will work on that point tomorrow. Sorry for the bug.

**Comment by RitchieP** [comment]
It's been quite a while since I worked on my project that required voice recording. I managed to make it work towards the end but I forgot what was the exact issue.

But I can share the link to my source code if you'd like to have a look.

https://github.com/RitchieP/VerbaLex/blob/main/verbalex/lib/utils/record_voice.dart

**Comment by Larpoux** [comment]
I am currently in the process of releasing a new flutter sound version with a bug fixed on Android with PCM16-WAV.
The new version will be accessible in half an hour.
I think that it will solve your issue.

**Comment by Larpoux** [comment]
Flutter Sound 9.16.1 is released : 

### 9.16.1

- Fix a bug on Android. The recorded file path was incorrect during StopRecorder() with codec=WAV-PCM16. [(973)](https://github.com/Canardoux/flutter_sound/issues/973).

**Comment by muneebahmad0600** [comment]
Issue fixed. Thanks @Larpoux. very much appreciated man

**Comment by gaohanas** [comment]
虽然在9.16.1这个版本上修复了这个问题，但是在最新的flutter_sound: ^9.27.0版本中这个问题依然存在。

**Comment by arazinger** [comment]
Yep, does not work in flutter_sound: ^9.28.0 either.

**Comment by Rostish** [comment]
Hi!
Faced the same problem with WAV-PCM16. aacMP4 works well.

**Comment by Larpoux** [comment]
This issue is a duplicate of issue #1187.
Please look to [this issue](https://github.com/Canardoux/flutter_sound/issues/1187) for more information.

[taudio 10.3.2](https://taudio.canardoux.xyz/) is released. The bug on Android when recording PCM16WAV is fixed.
It is really simple to upgrade from Flutter Sound 9.x. Just 5 minutes.
Here is a [small note](https://taudio.canardoux.xyz/tau/fs2taudio.html)
### 10.3.2

- Android : Remove dummy code when recording PCM16WAV ( [fs #1187](https://github.com/Canardoux/flutter_sound/issues/1187) )

**Comment by cuong292** [comment]
@Larpoux yes it is, flutter_sound depends on flutter_sound_web 9.28.0, but taudio depends on flutter_sound_web 9.30.0. So i can't upgrade it.

<img width="1384" alt="Image" src="https://github.com/user-attachments/assets/3df1d5c8-df44-4631-a238-b071b0bfc2e7" />

**Comment by Larpoux** [comment]
@cuong292 , your problem is because your pubspec.yaml depends both on flutter sound and Taudio. You must remove the dependency on flutter_sound
